modyn/supervisor/internal/triggers/performance/performancetrigger_mixin.py

```diff
@@ -64,22 +64,29 @@ def _inform_new_model(self, most_recent_model_id: int, last_detection_interval:
         self.most_recent_model_id = most_recent_model_id
         self.model_refresh_needed = True
 
+        assert last_detection_interval
+
         # Perform an evaluation of the NEW model on the last evaluation interval, we will derive expected performance
         # forecasts from these evaluations.
-        num_samples, num_misclassifications, evaluation_scores = self._run_evaluation(
-            interval_data=last_detection_interval
+        # num_samples, num_misclassifications, evaluation_scores = self._run_evaluation(
+        #     interval_data=last_detection_interval
+        # )
+        assert self.config.mode == "hindsight", (
+            "Forecasting mode is not supported yet, it requires tracking the performance right after trigger. "
+            "However, after triggers the models has learned from the last detection interval. We would need to "
+            "maintain a holdout set for this."
         )
```

Comment:
Is that (with the commented out code) what we should merge? Sorry, I am not fully sure yet about the changes here. Right now, this inform function does not do anything besides updating internal state

Reply:
Yep, I didn't want to remove it as it might still be useful in the future. The pre-mature implementation is meanly stripped, but I'd like to keep this interface abstraction.